File: 003/problem003.c
#include <stdio.h>
#include <math.h>

int isPrime(long number) {
    long i;
    for (i = 2 ; i < sqrt(number) ; i++) {
        if (number % i == 0 && i != number) {
            return 0;
        }
    }
    return 1;
}
int main(int argc, char *argv[]) {
    long num = 600851475143;
    long i;
    for (i = sqrt(num) ; i > 0 ; i--) {
        if (isPrime(i) && num % i  == 0) {
            printf("goal: %lu\n", i);
            break;
        }
    }
    return 0;
}

File: 003/problem003.go
package main

import (
    "fmt"
    "math"
)

func isPrime(num int) bool {
    for i := 2 ; i < int(math.Sqrt(float64(num))) ; i++ {
        if num % i == 0 {
            return false
        }
    }
    return true
}

func main() {
    num := 600851475143
    for i := int(math.Sqrt(float64(num))); i > 0 ; i-- {
        if isPrime(i) && num % i == 0 {
            fmt.Println(i)
            break
        }
    }
}
